### ISSUE TITLE
LiftGraph seems to depend on node traversal order

### ISSUE THREAD

**Issue by GZGavinZhao** [GZGavinZhao/autobuild#7]
Reproducer:
1. Clone GZGavinZhao/packages at branch `autobuild-repro`
2. `go run . query src:<path-to-packages> samba ldb tdb`. `autobuild` should say that there's a cycle.
3. Go to `utils/graph.go`. In the `LiftGraph` function, change this section:
```go
	for node := 0; node < len(adjMap); node++ {
		actualNode := node
		if choose(actualNode) {
			for adj := range adjMap[actualNode] {
				// println("Starting with", actualNode, "going to", adj)
				if err = liftDfs(adj, actualNode, choose, adjMap, visited, &res); err != nil {
					return
				}
			}
		}
	}
```
to
```go
	for node := 0; node < len(adjMap); node++ {
		actualNode := (node + 1597) % len(adjMap)
		if choose(actualNode) {
			for adj := range adjMap[actualNode] {
				// println("Starting with", actualNode, "going to", adj)
				if err = liftDfs(adj, actualNode, choose, adjMap, visited, &res); err != nil {
					return
				}
			}
		}
	}
```
4. Run `go run . query src:<path-to-packages> samba ldb tdb` again. `autobuild` should say the lifted graph doesn't contain cycles!

i.e. starting from `samba` works, but starting from `ldb` doesn't!

Why???

**Comment by GZGavinZhao** [comment]
I think this is fixed in 6b4a05f0b5fecdbdade10d021b58113977b38590 (exposed after 934df8f9cffb48661acd19221b7d9dc736d73027), but keeping this open for a while in case someone reports another nondeterministic issue.

**Comment by GZGavinZhao** [comment]
Seems to be fixed.
